File: java-solid-principle-sample-code/src/com/principle/solid/single_responsibility/AreaCalculator.java
package com.principle.solid.single_responsibility;

import java.util.List;

public class AreaCalculator {

	public int sum(List<Shape> shapes) {

		int sum = 0;
		for(int i = 0; i < shapes.size(); i++) {
			Shape shape = shapes.get(i);
			if(shape instanceof Square) {
				sum += Math.pow(((Square)shape).getLength(), 2);
			}
			if(shape instanceof Circle) {
				sum += Math.PI * Math.pow(((Circle)shape).getRadius(), 2);
			}
		}
		return sum;
	}
	
	// single responsibility principle violation
//	public String json(List<Shape> shapes) {
//		return String.format("{sum: %s}", sum(shapes));
//	}
}

File: python-solid-principle-sample-code/com/principle/solid/single_responsibility/AreaCalculator.py
from math import pi
from typing import List

from Circle import Circle
from Shape import Shape
from Square import Square


class AreaCalculator:

    def sum(self, shapes: List[Shape]):
        total_area = 0
        for shape in shapes:
            if isinstance(shape, Square):
                total_area += shape.getLength() ** 2
            if isinstance(shape, Circle):
                total_area += shape.getRadius() ** 2 * pi
        return total_area

    # single responsibility principle violation
    # def json(self, shapes):
    #     return "{{sum: {}}}".format(int(self.sum(shapes)))

